### PR TITLE
Refactors chapter 7

### PR DESCRIPTION
**Pull request by v-bpastiu**

### CODE REVIEW

File: 07-alerting-and-notification/README.md [hunk 1]
```diff
@@ -2,16 +2,21 @@
 
 ## Table of contents
```
Comment (v-ctiutiu) [inline 07-alerting-and-notification/README.md:3]: I would add a requirements or prerequisites section to this chapter as well. In it, you can refer to the Prometheus monitoring stack as the first requirement, then Emoji sample app, then Slack incoming webhook configuration, and any other requirement this chapter may have.

Don't forget to update TOC afterwards.

File: 07-alerting-and-notification/README.md [hunk 2]
```diff
@@ -39,80 +43,92 @@ The following is a sample set of alerts from that list.
...
+  HELM_CHART_VERSION="35.5.1"
 
-Often we need to be notified immediately about any critical issue in our cluster. That is where `AlertManager` comes into the picture. `Alertmanager` helps in aggregating the `alerts`, and sending notifications as shown in the diagram below.
+  helm upgrade kube-prom-stack prometheus-community/kube-prometheus-stack --version "${HELM_CHART_VERSION}" \
+    --namespace monitoring \
+    -f "04-setup-prometheus-stack/assets/manifests/prom-stack-values-v${HELM_CHART_VERSION}.yaml"
+  ```
 
-![AlertManager Filtering](assets/images/Prometheus-Notification-via-Slack.png)
+## Configuring Alertmanager to Send Notification to Slack
```
Comment (v-ctiutiu) [inline 07-alerting-and-notification/README.md:78]: Maybe:

> Configuring Alertmanager to Send Notifications to Slack

File: 07-alerting-and-notification/README.md [hunk 3]
```diff
@@ -39,80 +43,92 @@ The following is a sample set of alerts from that list.
...
 
-## Configuring alertmanager to send notification to Slack
+1. From the `Slack` desktop or web application, select the `Administration` --> `Manage Apps`.
+2. In the `Manage apps` directory, search for `Incoming WebHooks` in the search bar and click on the `Add to Slack button.
+3. In the `Post to Channel` section choose a channel you want AlertManager to send notifications to.
+4. On the next page, a `Webhook URL` will be displeyd. Make sure to copy it as you will use it in the next steps.
+5. Click on the `Save Settings` button at the bottom of the page.
 
-In your prom-stack-values.yaml file, modify the `config` part to include `slack_api_url`, `routes`, and `slack_config`.
+Next, we will add the necessary config to `AlertManager` to enable `Slack` notifications. Open the `04-setup-prometheus-stack/assets/manifests/prom-stack-values.yaml` file provided in the `Starter Kit` repository, using a text editor of your choice (preferably with `YAML` lint support). Uncomment the `config` block. Make sure to update the `slack_api_url` and `channel` values.
```
Comment (v-ctiutiu) [inline 07-alerting-and-notification/README.md:90]: Try not to use the "we".

File: 07-alerting-and-notification/README.md [hunk 4]
```diff
@@ -2,16 +2,21 @@
...
+- [Alerting and Notification Overview](#alerting-and-notification-overview)
+- [List of Included Alerts](#list-of-included-alerts)
+- [Creating a New Alert](#creating-a-new-alert)
+- [Configuring Alertmanager to Send Notification to Slack](#configuring-alertmanager-to-send-notification-to-slack)
 
-## Overview
+## Alerting and Notification Overview
 
-Alerts and Notifications are a critical part of your day 2 workflow. When things go wrong (eg. any service is down, or a pod is crashing etc.), you expect to see the appropriate alerts, and notifications to handle these situations. The Starter Kit includes a set of default alerts as part of the `kube-prom-stack` installation.
+Often times we need to be notified immediately about any critical issue in our cluster. That is where `AlertManager` comes into the picture. `Alertmanager` helps in aggregating the `alerts`, and sending `notifications` as shown in the diagram below.
```
Comment (v-ctiutiu) [inline 07-alerting-and-notification/README.md:12]: Try not to use the "we".

File: 07-alerting-and-notification/README.md [hunk 5]
```diff
@@ -39,80 +43,92 @@ The following is a sample set of alerts from that list.
...
 
-```shell
-alertmanager:  
+```yaml
+alertmanager:
+  enabled: true
   config:
     global:
       resolve_timeout: 5m
       slack_api_url: "https://hooks.slack.com/services/TXXXX/BXXXX/<token>"
```
Comment (v-ctiutiu) [inline 07-alerting-and-notification/README.md:98]: You can use a `<YOUR_SLACK_APP_INCOMING_WEBHOOK_URL_HERE>` placeholder here.

**Review by v-ctiutiu — CHANGES_REQUESTED** [review]

File: 07-alerting-and-notification/README.md [hunk 6]
```diff
@@ -39,80 +43,92 @@ The following is a sample set of alerts from that list.
...
+  HELM_CHART_VERSION="35.5.1"
+
+  helm upgrade kube-prom-stack prometheus-community/kube-prometheus-stack --version "${HELM_CHART_VERSION}" \
+    --namespace monitoring \
+    --create-namespace \
+    -f "04-setup-prometheus-stack/assets/manifests/prom-stack-values-v${HELM_CHART_VERSION}.yaml"
+  ```
 
-At this point, you should see the slack `notifications` for firing alerts.
+At this point, you should see the slack `notifications` for firing alerts when the expected number of pods for `Emojivoto` is less than 4.
```
Comment (v-ctiutiu) [inline 07-alerting-and-notification/README.md:132]: You are giving an example for a scenario where the emojivoto application pods count is less than 4. But, I don't see how you trigger this situation. Is there a step missing where you are downscaling the application to let's say 3 pods to simulate this scenario ?

File: 07-alerting-and-notification/README.md [hunk 7]
```diff
@@ -39,80 +43,92 @@ The following is a sample set of alerts from that list.
...
-helm upgrade kube-prom-stack prometheus-community/kube-prometheus-stack --version 30.0.1 --namespace monitoring --create-namespace -f prom-stack-values.yaml
+Finally, upgrade the `kube-prometheus-stack`, using `Helm`:
 
-```
+  ```shell
+  HELM_CHART_VERSION="35.5.1"
+
+  helm upgrade kube-prom-stack prometheus-community/kube-prometheus-stack --version "${HELM_CHART_VERSION}" \
+    --namespace monitoring \
+    --create-namespace \
```
Comment (v-ctiutiu) [inline 07-alerting-and-notification/README.md:128]: You don't need the `--create-namespace` flag in this case.

File: 07-alerting-and-notification/README.md [hunk 8]
```diff
@@ -39,80 +52,109 @@ The following is a sample set of alerts from that list.
...
-- *routes:* in this part, you will have some `receivers` and tell `routes` for the `matching receiver` here.
-- *slack_configs:* this segment includes more than 4 parts but here, `channel`, `title`, `text` are important parts for telling `Slack router` which channel and which title-text message will have. This part is totally related to how to use `notifications` in `prom-stack-values.yaml`. [Please visit the Prometheus-Notifications page for more details about this chart](https://prometheus.io/docs/alerting/latest/notification_examples/).
+Explanations for the above configuration:
 
-Now re-apply the new `prom-stack-values.yaml` using helm upgrade.Apply below upgrading command:
+- `slack_api_url` - url of the `webhook` created in step 4.
+- `routes.match.alertname` - name of the `AlertManager` alert.
+- `slack_configs` - configuration related to the channel to send the notification to, the title and the message it should include.
```
Comment (v-ctiutiu) [inline 07-alerting-and-notification/README.md:129]: > `receivers.[].slack_configs` - defines the Slack channel used to send notifications, notification title and the actual message. It is also possible to format the notification message (or body) based on your requirements.

File: 07-alerting-and-notification/README.md [hunk 9]
```diff
@@ -39,80 +52,109 @@ The following is a sample set of alerts from that list.
...
-```
+  ```shell
+  HELM_CHART_VERSION="35.5.1"
+
+  helm upgrade kube-prom-stack prometheus-community/kube-prometheus-stack --version "${HELM_CHART_VERSION}" \
+    --namespace monitoring \
+    -f "04-setup-prometheus-stack/assets/manifests/prom-stack-values-v${HELM_CHART_VERSION}.yaml"
+  ```
+
+At this point, you should see the slack `notifications` for firing alerts when the expected number of pods for `Emojivoto` is less than 4.
```
Comment (v-ctiutiu) [inline 07-alerting-and-notification/README.md:141]: Maybe:

> At this point, you should receive slack notifications for firing alerts when the expected number of pods for `Emojivoto` is less than 4.

File: 07-alerting-and-notification/README.md [hunk 10]
```diff
@@ -39,80 +52,109 @@ The following is a sample set of alerts from that list.
...
+Next, you will be testing the firing of the `Alert` and the sending of the `Slack` notification by downscaling the number of replicas for a `deployment` in the `emojivoto` namespace.
```
Comment (v-ctiutiu) [inline 07-alerting-and-notification/README.md:142]: Be more concise:

> Next, you're going to test if the `EmojivotoInstanceDown` alert works by downscaling the number of replicas for the `emoji` deployment from the `emojivoto` namespace.

**Review by v-ctiutiu — CHANGES_REQUESTED** [review]

File: 04-setup-prometheus-stack/assets/manifests/prom-stack-values-v35.5.1.yaml [hunk 11]
```diff
@@ -120,16 +153,16 @@ prometheus:
   # Starter Kit setup for DigitalOcean Block Storage
```
[the remark this answers is not in the document]
Comment (v-bpastiu) [reply 04-setup-prometheus-stack/assets/manifests/prom-stack-values-v35.5.1.yaml:153]: I can remove it.

**Review by v-ctiutiu — APPROVED** [review]
Looks good to me now.

**Pull request merged**